great_expectations/rule_based_profiler/types/domain.py

```diff
@@ -124,13 +126,17 @@ def to_json_dict(self) -> dict:
         details: dict = {}
 
         key: str
-        semantic_type: Union[str, SemanticDomainTypes]
-        for key, semantic_type in self["details"].items():
-            if isinstance(semantic_type, str):
-                semantic_type = semantic_type.lower()
-                semantic_type = SemanticDomainTypes(semantic_type)
-
-            details[key] = semantic_type.value
+        value: Any
+        for key, value in self["details"].items():
+            if key == INFERRED_SEMANTIC_TYPE_KEY:
+                semantic_type: Union[str, SemanticDomainTypes]
+                if isinstance(value, str):
+                    semantic_type = value.lower()
+                    semantic_type = SemanticDomainTypes(semantic_type)
+                else:
+                    semantic_type = value
+
+                details[key] = semantic_type.value
```

Comment:
Should we also do the else clause so we don't lose non semantic type details?:
```suggestion
        value: Any
        for key, value in self["details"].items():
            if key == INFERRED_SEMANTIC_TYPE_KEY:
                semantic_type: Union[str, SemanticDomainTypes]
                if isinstance(value, str):
                    semantic_type = value.lower()
                    semantic_type = SemanticDomainTypes(semantic_type)
                else:
                    semantic_type = value

                details[key] = semantic_type.value
             else:
                 details[key] = value
```

Reply:
@anthonyburdi great catch -- in next commit -- thanks!